### PR TITLE
[18.0][MIG] website_user_login_redirect_custom: forward port from 14.0

### PR DESCRIPTION
**Pull request by Mohajiro**
Forward port the following PR:

[[14.0][ADD] website_user_login_redirect_custom: redirect website users to custom URL](https://github.com/OCA/website/pull/1138/commits/92821e38a72ca14d94099b3e713d40696fb0abfa)

Task: 5058

## Summary by CodeRabbit

* **New Features**
  * Custom redirect after website/portal user login/signup with a settings UI to enable it and specify a relative URL

* **Behavior & Validation**
  * URL validation, trimming and normalization to ensure safe, well-formed relative redirect paths; sensible defaults added

* **Documentation**
  * README, CONFIGURE/USAGE/DESCRIPTION/CONTEXT/CONTRIBUTORS and a static module description page

* **Tests**
  * End-to-end tests covering enablement, validation, normalization, and edge cases

* **Chores**
  * Module manifest and packaging files added

### CODE REVIEW

File: website_user_login_redirect_custom/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        # Apply our custom redirect logic
+        if self._should_redirect_custom():
+            # Only apply custom redirect for cases where user would go to /my
+            # or when there's no specific redirect
+            if not parent_redirect or "/my" in parent_redirect:
+                custom_url = self._get_custom_redirect_url()
+                if custom_url:
+                    return custom_url
+
+        return parent_redirect
```
[the remark this answers is not in the document]
Comment (Mohajiro) [reply website_user_login_redirect_custom/controllers/main.py:64]: This is migration PR, so let's keep the same logic, the basic module was checked by good reviewers

File: website_user_login_redirect_custom/tests/test_website_login_redirect.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                "groups_id": [(4, cls.env.ref("base.group_user").id)],
+            }
+        )
+        cls.portal_user = cls.env["res.users"].create(
+            {
+                "name": "Portal User",
+                "login": "portal@test.com",
+                "groups_id": [(4, cls.env.ref("base.group_portal").id)],
+            }
+        )
```
[the remark this answers is not in the document]
Comment (Mohajiro) [reply website_user_login_redirect_custom/tests/test_website_login_redirect.py:31]: This is migration PR, let's keep the current tests.

File: website_user_login_redirect_custom/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[build-system]
+requires = ["whool"]
+build-backend = "whool.buildapi"
```
[the remark this answers is not in the document]
Comment (Mohajiro) [reply website_user_login_redirect_custom/pyproject.toml:3]: This is auto-generated file so I don't want to modify it
